wire-ios-request-strategy/Sources/Object Syncs/Helpers/MLSMessageSync.swift

```diff
@@ -83,7 +83,6 @@ class MLSMessageSync<Message: MLSMessage>: NSObject, ZMContextChangeTrackerSourc
             context.performGroupedBlock { [dependencySync] in
                 dependencySync.synchronize(entity: message, completion: completion)
                 RequestAvailableNotification.notifyNewRequestsAvailable(nil)
```

Comment:
should this (RequestAvailableNotification) be called within the passed completion? looks weird too

Reply:
no, its needed to generate a request. The completion should be called in case of a failure or a success response.

Reply:
I wanted to know what's the reason for generating a request here

Reply:
To send a message you need to:
- Insert a new `ZMClientMessage` in the database
- "Sync" the message via `MessageSync`. This will in turn forward it to either `ProteusMessageSync` or `MLSMessageSync` depending on the conversation.
- This will forward it to a "dependency sync" which will generate one or more requests depending on the dependencies of the message (eg. maybe we need to fetch conversation or user metadata first before we can send the message)
- Start the operation loop polling by posting the request available notification: this will ask the `ClientMessageRequestStrategy` if it needs to make any request, which will ask message sync if it needs to make requests, which will ask the dependency sync if it needs to make requests, which will eventually generate the request to send the message.
- When generating the request to send the message, we encrypt the message, put it in a payload, and send it to the backend.
- We handle the response and update local state if needed (e.g mark the message as being sent).

Reply:
thks I got it now - also looking at DependencyEntitySync help :)